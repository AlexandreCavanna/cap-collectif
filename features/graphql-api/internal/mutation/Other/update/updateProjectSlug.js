/* eslint-env jest */
import '../../../../_setup';

const UpdateProjectSlug = /* GraphQL*/ `
    mutation UpdateProjectSlugMutation(
        $input: UpdateProjectSlugInput!
    ) {
        updateProjectSlug(input: $input) {
            project {
                slug
                url
            }
            errorCode
        }
    }
`;

const input = {
  projectId: 'UHJvamVjdDpwcm9qZWN0MQ==',
  slug: 'toto 1234 éééà',
};

describe('mutations.createEvent', () => {
  it('should update slug', async () => {
    await expect(
      graphql(
        UpdateProjectSlug,
        {
          input,
        },
        'internal_admin',
      ),
    ).resolves.toMatchSnapshot();
  });
  it('should return PROJECT_NOT_FOUND errorCode', async () => {
    await expect(
      graphql(
        UpdateProjectSlug,
        {
          input: {
            ...input,
            projectId: 'abc',
          },
        },
        'internal_admin',
      ),
    ).rejects.toThrowError('Access denied to this field.');
  });
});
