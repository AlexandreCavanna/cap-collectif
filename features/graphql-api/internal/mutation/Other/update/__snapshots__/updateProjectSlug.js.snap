// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mutations.createEvent should update slug 1`] = `
Object {
  "updateProjectSlug": Object {
    "errorCode": null,
    "project": Object {
      "slug": "toto-1234-eeea",
      "url": "https://capco.test/project/toto-1234-eeea/presentation/presentation-1",
    },
  },
}
`;
